oauth/api/router.py

```diff
@@ -0,0 +1,345 @@
+# pylint: disable=unused-argument
+
+from datetime import datetime
+from json import loads as load
+from logging import getLogger
+from typing import Any, Callable, Generator, List, Optional
+
+from fastapi import APIRouter, Body, Depends, Request, Security, status
+from fastapi.routing import APIRoute
+from fastapi.security import HTTPBasic, SecurityScopes
+from sqlalchemy.orm import Session
+
+import oauth.api.endpoints as endpoints
+import oauth.jwt as jwt
+from oauth.api.exceptions import (
+    AuthenticationException,
+    AuthorizationException,
+    ClientNotFoundException,
+    ExpiredTokenException,
+)
+from oauth.api.models import (
+    AccessToken,
+    ClientCreatedResponse,
+    OAuth2ClientCredentialsBearer,
+    OAuth2ClientCredentialsRequestForm,
+)
+from oauth.api.utils import extract_payload, is_token_expired, validate_scopes
+from oauth.database.models import ClientDetail
+from oauth.scopes import (
+    CLIENT_CREATE,
+    CLIENT_DELETE,
+    CLIENT_READ,
+    CLIENT_UPDATE,
+    SCOPE_DOCS,
+    SCOPE_READ,
+    SCOPES,
+)
+
+EIGHT_DAYS = 60 * 24 * 8  # Expressed in minutes
+SORTED_SCOPES = SCOPES.sort()
+
+logger = getLogger(__name__)
+oauth2_scheme = OAuth2ClientCredentialsBearer(
+    tokenUrl=endpoints.TOKEN,
+    scopes=SCOPE_DOCS,
+)
+
+
+class OAuthRouter(APIRouter):
```

Comment:
Did this mega class end up being easier than creating some kind of `get_oauth_router` function that we had talked about?

Reply:
It's a personal preference. Functional style would work as well, but I think it provides a more intuitive user experience for the caller to be able to do:
```python
router.include_router(OAuthRouter(...))
```
It's "ok, I'm creating an instance of a thing" vs "I'm calling some function and 🤞".